lib/spack/spack/build_environment.py

```diff
@@ -438,6 +439,15 @@ def _set_variables_for_single_module(pkg, module):
     m.cmake = Executable('cmake')
     m.ctest = MakeExecutable('ctest', jobs)
 
+    m.cargo = Executable('cargo')
+    m.cargo.add_default_arg('--locked')
+    m.cargo.add_default_arg('--offline')
+
+    m.cargo_build = copy.deepcopy(m.cargo)
+    m.cargo_build.add_default_arg('build')
+    m.cargo_build.add_default_arg('--jobs')
+    m.cargo_build.add_default_arg(str(jobs))
```

Comment:
A lot of the stuff in this function is leftover from before Spack had base classes. I would put this stuff in `CargoPackage`, or in the `cargo` package class. We handle `python` and `bazel` similarly I think.